matching-notification-api/src/main/java/org/phenotips/matchingnotification/internal/DefaultMatchingNotificationManager.java

```diff
@@ -0,0 +1,252 @@
+import org.phenotips.matchingnotification.finder.MatchFinderManager;
+import org.phenotips.matchingnotification.match.PatientMatch;
+import org.phenotips.matchingnotification.match.internal.DefaultPatientMatch;
+import org.phenotips.matchingnotification.notification.PatientMatchEmail;
+import org.phenotips.matchingnotification.notification.PatientMatchNotificationResponse;
+import org.phenotips.matchingnotification.notification.PatientMatchNotifier;
+import org.phenotips.matchingnotification.storage.MatchStorageManager;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryManager;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Joiner;
+
+/**
+ * @version $Id$
+ */
+@Component
+@Singleton
+public class DefaultMatchingNotificationManager implements MatchingNotificationManager
+{
+    private Logger logger = LoggerFactory.getLogger(DefaultMatchingNotificationManager.class);
+
+    @Inject
+    private QueryManager qm;
+
+    @Inject
+    private PatientRepository patientRepository;
+
+    @Inject
+    private PermissionsManager permissionsManager;
+
+    @Inject
+    @Named("matchable")
+    private Visibility matchableVisibility;
+
+    @Inject
+    private PatientMatchNotifier notifier;
+
+    @Inject
+    private MatchFinderManager matchFinderManager;
+
+    @Inject
+    private MatchStorageManager matchStorageManager;
+
+    @Override
+    public List<PatientMatch> findAndSaveMatches(double score)
+    {
+        List<PatientMatch> addedMatches = new LinkedList<>();
+
+        List<Patient> patients = this.getPatientsList();
+        for (Patient patient : patients) {
+
+            this.logger.debug("Finding matches for patient {}.", patient.getId());
+
+            List<PatientMatch> matchesForPatient = this.matchFinderManager.findMatches(patient);
+            this.filterMatchesByScore(matchesForPatient, score);
+            this.filterExistingMatches(matchesForPatient);
+            this.matchStorageManager.saveMatches(matchesForPatient);
+
+            addedMatches.addAll(matchesForPatient);
+        }
+
+        return addedMatches;
+    }
+
+    /*
+     * Returns a list of patients with visibility>=matchable
+     */
+    private List<Patient> getPatientsList()
+    {
+        List<String> potentialPatientIds = null;
+        List<Patient> patients = new LinkedList<>();
+        try {
+            Query q = this.qm.createQuery(
+                "select doc.name " + "from Document doc, " + "doc.object(PhenoTips.PatientClass) as patient "
```

Comment:
I usually write it in different lines, and eclipse must have removed the line breaks.